python/taichi/lang/struct.py

```diff
@@ -76,7 +76,15 @@ def __setitem__(self, key, value):
             self.entries[key].accessor.setter(value, *self.entries[key].key)
         else:
             if in_python_scope():
-                self.entries[key].set_entries(value)
+                if isinstance(self.entries[key], Struct) or isinstance(self.entries[key], Matrix):
+                    self.entries[key].set_entries(value)
+                else:
+                    if isinstance(value, (numbers.Number, expr.Expr)):
```

Comment:
```suggestion
                    if isinstance(value, numbers.Number):
```
`expr.Expr` is used in Taichi scope. Since we already checked the method is called in Python scope, we can just verify whether it is a `Number` object.

Reply:
OK, I see. Thanks a lot!